test/test_respondent_controller.py

```diff
@@ -680,6 +698,67 @@ def test_post_add_survey_no_business_raise_ras_error(self):
+                                               respondent_id=self.mock_respondent_with_id['id'])
+        self.populate_with_enrolment()
+        db_respondent = respondents()[0]
+        token = self.generate_valid_token_from_email(db_respondent.email_address)
+        self.put_email_verification(token, 200)
+        request_json = {
+            'respondent_party_id': self.mock_respondent_with_id['id'],
+            'business_id': '3b136c4b-7a14-4904-9e01-13364dd7b972',
+            'survey_id': '02b9c366-7397-42f7-942a-76dc5876d86d',
+            'change_flag': 'DISABLED'
+        }
+        self.put_enrolment_status(request_json, 200)
+
+    def test_put_change_respondent_enrolment_status_enabled_success(self):
+        def mock_put_iac(*args, **kwargs):
+            return MockResponse('{"active": false}')
+        self.mock_requests.put = mock_put_iac
+        self.populate_with_respondent(respondent=self.mock_respondent_with_id)
+        self.populate_with_business()
+        self.associate_business_and_respondent(business_id='3b136c4b-7a14-4904-9e01-13364dd7b972',
+                                               respondent_id=self.mock_respondent_with_id['id'])
+        enrolment = self.mock_enrolment_disabled
+        self.populate_with_enrolment(enrolment=enrolment)
+        db_respondent = respondents()[0]
+        token = self.generate_valid_token_from_email(db_respondent.email_address)
+        self.put_email_verification(token, 200)
+        request_json = {
+            'respondent_party_id': self.mock_respondent_with_id['id'],
+            'business_id': '3b136c4b-7a14-4904-9e01-13364dd7b972',
+            'survey_id': '02b9c366-7397-42f7-942a-76dc5876d86d',
+            'change_flag': 'ENABLED'
+        }
+        self.put_enrolment_status(request_json, 200)
+
+    def test_put_change_respondent_enrolment_status_random_string_fail(self):
+        def mock_put_iac(*args, **kwargs):
+            return MockResponse('{"active": false}')
+        self.mock_requests.put = mock_put_iac
+        self.populate_with_respondent(respondent=self.mock_respondent_with_id)
+        self.populate_with_business()
+        self.associate_business_and_respondent(business_id='3b136c4b-7a14-4904-9e01-13364dd7b972',
+                                               respondent_id=self.mock_respondent_with_id['id'])
+        self.populate_with_enrolment()
+        db_respondent = respondents()[0]
+        token = self.generate_valid_token_from_email(db_respondent.email_address)
+        self.put_email_verification(token, 200)
+        request_json = {
+            'respondent_party_id': self.mock_respondent_with_id['id'],
+            'business_id': '3b136c4b-7a14-4904-9e01-13364dd7b972',
+            'survey_id': '02b9c366-7397-42f7-942a-76dc5876d86d',
+            'change_flag': 'woafouewbhouGFHEPIW0'
+        }
+        self.put_enrolment_status(request_json, 500)
+
```

Comment:
I think it needs a test for if the respondent we are trying to change the enrolment status of doesn't exist.

Reply:
+1

Reply:
Will add this in :+1: